device-connectors/src/testflinger_device_connectors/devices/__init__.py

```diff
@@ -54,11 +54,27 @@
 
 
 class ProvisioningError(Exception):
-    pass
+    def __init__(self, message: str = None):
+        if message is not None:
+            with open(
+                "provision_error.log", "w", encoding="utf-8"
+            ) as provision_error_file:
+                provision_error_file.write(message)
+                super().__init__(message)
+        else:
+            super().__init__()
```

Comment:
I don't think I would handle writing the error out in the exception class itself. Instead, catch the error up at a higher level and handle the write once, along with the type of error, etc. It might make better sense to save it as a json file so that you can be flexible and capture more than just a single detailed err - then you could specify the type of error, and other details as needed? Just a thought.